### DISCUSSION TITLE
check fills given an OrderId

### DISCUSSION THREAD

**Discussion by richy1996** [hootnot/saxo_openapi#30] [Q&A]
how to check fills given an OrderId? Seems AverageFillPrice is something that can be received but I'm not sure how using this wrapper. info here https://www.developer.saxo/openapi/learn/order-events

**Comment by hootnot** [comment]
I guess one way is `trading / messages`. If an order gets filled or filled for a part, messages will be sent. You can 'GET' them, but you also can subscribe for them. So, when you have a subscription and parse the messages you receive, I guess you should be able to extract the details you want.

```
r = tr.messages.GetTradeMessages()
client.request(r)
```

or the subscription: ```CreateTradeMessageSubscription(...)```

Processing messages from a subscription allows you to process the messages instantly. 
(not sure if all info you want is in that message)

**Reply by richy1996** [reply]
I think the endpoint I'm looking for is here https://www.developer.saxo/openapi/learn/audit-orderactivities. I was able to get it to work by running the below

```python
url =  f'https://gateway.saxobank.com/sim/openapi/cs/v1/audit/orderactivities/?OrderId={order_id}&EntryType=All'

headers = {
    'Authorization': f'Bearer {tok}',
    'Content-Type': 'application/json',
}

response = requests.get(url=url, headers=headers)

response.json()
...
```

Which brings up another question for me, where does the documentation say what headers to put in the http request? E.g. how do I know to put this
```python
headers = {
    'Authorization': f'Bearer {tok}',
    'Content-Type': 'application/json',
}
```

Thankfully chatgpt is around. I tried to figure out some headers by debugging your wrapper but I can't understand that god tier python.

**Reply by hootnot** [reply]
headers: https://www.developer.saxo/openapi/learn/openapi-request-response

Using saxo_openapi, headers is taken care of. 

If you have specific questions regarding their API, pls. drop them at the SAXO-developer helpdesk.
